### PR TITLE
Activate Zenoh logs by default

### PR DESCRIPTION
**Pull request by JEnoch**
Currently no logs from Zenoh are shown, neither "error" level, neither "warn" level.
The user would have to define the `RUST_LOG` environment variable to enable those logs.

This PR checks automatically defines the `RUST_LOG` environment variable before opening a Zenoh Session, if not already defined in the environment.
- For the Router it's set by default to `RUST_LOG=info` (similar level than the original Zenoh Router)
- For the Nodes it's set by default to `RUST_LOG=warn`

### CODE REVIEW

**Comment by JEnoch** [comment]
This is a follow up of [this discussion](https://github.com/ros2/rmw_zenoh/issues/238#issuecomment-2211175116)

File: rmw_zenoh_cpp/src/rmw_init.cpp [hunk 1]
```diff
@@ -42,6 +42,10 @@ extern "C"
 // TODO(clalancette): Make this configurable, or get it from the configuration
 #define SHM_BUFFER_SIZE_MB 10
 
+// Environment variable for Zenoh logging
+const char * ZENOH_LOG_ENV_VAR = "RUST_LOG";
```
Comment (Yadunund) [inline rmw_zenoh_cpp/src/rmw_init.cpp]: My preference would be to move these definitions into `detail/zenoh_config.hpp` which is already included by `rmw_init.cpp` and `zenohd/main.cpp` below. Then you can define a `ZENOH_LOG_INFO_LEVEL_STR` and `ZENOH_LOG_DEBUG_LEVEL_STR` and use these in the `setenv` call.

**Review by Yadunund — CHANGES_REQUESTED** [review]
Thanks for updating the implementation! Please find my feedback below.

File: rmw_zenoh_cpp/src/rmw_init.cpp (hunk 1 above)
Comment (JEnoch) [reply rmw_zenoh_cpp/src/rmw_init.cpp]: Done!

**Review by Yadunund — APPROVED** [review]
Pushed a small change to return an `rmw_ret_t` enum in `rmw_init` but otherwise LGTM!

**Pull request merged**